ts/receiver/cache.ts

```diff
@@ -1,7 +1,103 @@
 import { EnvelopePlus } from './types';
+import { StringUtils } from '../session/utils';
+import { toNumber } from 'lodash';
 
-export function removeFromCache(envelope: EnvelopePlus) {
+export async function removeFromCache(envelope: EnvelopePlus) {
   const { id } = envelope;
 
   return window.textsecure.storage.unprocessed.remove(id);
 }
+
+export async function addToCache(
+  envelope: EnvelopePlus,
+  plaintext: ArrayBuffer
+) {
+  const { id } = envelope;
+
+  const encodedEnvelope = StringUtils.decode(plaintext, 'base64');
+  const data: any = {
+    id,
+    version: 2,
+    envelope: encodedEnvelope,
+    timestamp: Date.now(),
+    attempts: 1,
+  };
+
+  if (envelope.senderIdentity) {
+    data.senderIdentity = envelope.senderIdentity;
+  }
+
+  return window.textsecure.storage.unprocessed.add(data);
+}
+
+export async function getAllFromCache() {
+  window.log.info('getAllFromCache');
+
+  const { textsecure, Lodash: _ } = window;
```

Comment:
lodash can be imported directly in ts
```
import _ from 'lodash'
```

Reply:
Yep, missed this one. Updated this one and in a few other places.